lib/src/metta/runner/stdlib.rs

```diff
@@ -61,101 +47,79 @@ impl Display for ImportOp {
 
 impl Grounded for ImportOp {
     fn type_(&self) -> Atom {
+        //QUESTION: How do we express that we can accept a variable number of arguments?
+        // Ideally the "import as" / "import into" part would be optional
```

Comment:
Yes, I'm simulating variable number of arguments wrapping them into `Expression`. Not sure if non-deterministic types is a good idea here. They produce additional results sometimes

Reply:
> they will produce additional results if arguments can be casted to more than one type.

If we take any single Atom as the arg, and then do type checking inside the grounded atom's execute function that seems workable.

Are you saying the grounded atom's execute function might be invoked multiple times?  Because if that were true that would also be true for any signature.

@vsbogd @Necr0x0Der Do you think we should change the `import!` signature now?  Since changing it later will have downstream effects on all users?

Or should we wait until after alpha / next-breaking-revision to see where we stand and decide on how we want to support more import options?

Reply:
> Are you saying the grounded atom's execute function might be invoked multiple times? Because if that were true that would also be true for any signature.

It is true for any function grounded or not. It works like I wrote it is called many times if arguments can be casted to more than one type.

Reply:
> @vsbogd @Necr0x0Der Do you think we should change the import! signature now? Since changing it later will have downstream effects on all users?

Am I right this PR already changes a signature (args of import! are swapped)? Do we want to keep it or swap them back before merging? If we change signature what kind of imports will be supported automatically?

Reply:
Yes, args of `import!` are swapped, so the signature is changed. If we are thinking about extended API for `import!`, it would possibly be better to make the signature suitable for further changes or to undo this change for backward compatibility (if we are thinking about changing it again, we should avoid two changes, and both of these approaches can satisfy this).
One question I have is about evaluating functions in the context of the imported space as well as accessing tokens of the imported module. Depending on this, we either need a way to assign names to imported modules or not.
I don't like too much how it was done originally. It was convenient for a narrow use, but it raises some questions. Currently, `import!` adds a new token if the argument is not `Space`. It's convenient for some cases, but I'd prefer `import` to just return a space in others (although it wouldn't work well for modules). it's also a little bit weird for namespacing, especially if we want to use tokens defined in another module. If a module has both tokens and non-empty space content, they if we import it to another space, we lose access to tokens. If we import it to `&self`, we mix up its content with `&self`. That's why I proposed to separate imports of spaces from imports of modules. Module is a space with a tokenizer, or, more generally, is a runner. We can have, say, rholang as a whole runner - not just space. And we already have something like this for metta-morph, although in a custom ad hoc way. We need something like Runner API besides Space API...
As for,
```
!(import! (<name> into <space>)) ; import specific name into space
!(import! (<name> as <alias> into <space>)) ; to import into space as alias
!(import! <name>) ; to import into self
!(import! (<name> as <alias>)) ; to import into self as alias
```
I don't like `into` too much as it might be excessive. I don't like `!(import! <name>)` too much, because importing a space to `&self` shouldn't be a good practice in general. And I'm not sure how `alias` should be used. And we go back to the question about use cases. We need extensions typically for additional tokens, which we'd like to access easily. Sometimes, with spaces. We can have a special mechanism for importing tokens, e.g. `(from das_gate import new-das)` or `(from das_gate import *)` . But it will be confusing that we cannot import normal functions from modules this way. We can have `import!` and `extend!`, which behave differently (`extend!` will import all tokens to the caller module even if the module is imported as a separate space). Extensions being imported to `&self` can actually create a space with their additional code in any case and provide access to this code space via a special token they create by themselves. But it would be better handled by the module system in a uniform way, I think. 

Thus, there is a lot to discuss. I wouldn't want these considerations and issues to be a blocker for merging this PR. Maybe, we need to just swap arguments of `import!` back, and do the next major revision of imports before alpha release.